valhalla/tyr/actor.h

```diff
@@ -13,32 +14,40 @@ namespace tyr {
+  actor_t(const boost::property_tree::ptree& config,
+          baldr::GraphReader& reader,
+          bool auto_cleanup = false);
   void cleanup();
   std::string route(const std::string& request_str,
-                    const std::function<void()>& interrupt = []() -> void {});
+                    const std::function<void()>& interrupt = []() -> void {},
+                    Api* api = nullptr);
   std::string locate(const std::string& request_str,
-                     const std::function<void()>& interrupt = []() -> void {});
+                     const std::function<void()>& interrupt = []() -> void {},
+                     Api* api = nullptr);
   std::string matrix(const std::string& request_str,
-                     const std::function<void()>& interrupt = []() -> void {});
+                     const std::function<void()>& interrupt = []() -> void {},
+                     Api* api = nullptr);
   std::string optimized_route(const std::string& request_str,
-                              const std::function<void()>& interrupt = []() -> void {});
+                              const std::function<void()>& interrupt = []() -> void {},
+                              Api* api = nullptr);
   std::string isochrone(const std::string& request_str,
-                        const std::function<void()>& interrupt = []() -> void {});
+                        const std::function<void()>& interrupt = []() -> void {},
+                        Api* api = nullptr);
   std::string trace_route(const std::string& request_str,
-                          const std::function<void()>& interrupt = []() -> void {});
+                          const std::function<void()>& interrupt = []() -> void {},
+                          Api* api = nullptr);
   std::string trace_attributes(const std::string& request_str,
-                               const std::function<void()>& interrupt = []() -> void {});
+                               const std::function<void()>& interrupt = []() -> void {},
+                               Api* api = nullptr);
   std::string height(const std::string& request_str,
-                     const std::function<void()>& interrupt = []() -> void {});
+                     const std::function<void()>& interrupt = []() -> void {},
+                     Api* api = nullptr);
   std::string transit_available(const std::string& request_str,
-                                const std::function<void()>& interrupt = []() -> void {});
+                                const std::function<void()>& interrupt = []() -> void {},
+                                Api* api = nullptr);
   std::string expansion(const std::string& request_str,
-                        const std::function<void()>& interrupt = []() -> void {});
-
-  valhalla::Api unserialized_route(const std::string& request_str,
-                                   const std::function<void()>& interrupt = []() -> void {});
-  valhalla::Api unserialized_trace_route(const std::string& request_str,
```

Comment:
i removed the unserialized functions in favor of adding the api pointer to all of the functions. so if you pass a non null Api object to the function the actor will pass it back to you at that pointer.